File: Binary Search/BinarySearchIterative.java
import java.util.*;

class BinarySearchIterative {
    public static int binarySearch(int[] array, int target) {
        // Write your code here.
        return binarySearch(array, target, 0, array.length - 1);
    }

    public static int binarySearch(int[] array, int target, int left, int right) {
        int pivot;
        while (left <= right) {
            pivot = (left + right) / 2;
            if (array[pivot] == target) {
                return pivot;
            } else if (array[pivot] < target) {
                left = pivot + 1;
            } else {
                right = pivot - 1;
            }
        }
        return -1;
    }
}

File: Binary Search/BinarySearchIterative.kt
package com.algoexpert.program

fun binarySearch(array: List<Int>, target: Int): Int {
    // Write your code here.
    return binarySearch(array, target, 0, array.size - 1)
}

fun binarySearch(array: List<Int>, target: Int, left: Int, right: Int): Int {
    var pivot: Int
    var l = left
    var r = right
    while (l <= r) {
        pivot = (l + r) / 2
        if (array[pivot] == target) {
            return pivot
        } else if (array[pivot] < target) {
            l = pivot + 1
        } else {
            r = pivot - 1
        }
    }
    return -1
}
